core/stream/format.go

```diff
@@ -129,6 +129,21 @@ func (f *Format) HasColorComponent() bool {
 	return false
 }
 
+// GetSingleColorComponent returns a component if the format contains a single
+// color component, otherwise nil. See ColorChannels for channels considered colors.
+func (f *Format) GetSingleColorComponent() *Component {
```

Comment:
That is not what it does, though, nor what I want it to do. I want it to check whether the format represents a single color channel, not a format that has at least one color channel.